clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -420,6 +420,10 @@ def err_drv_fsycl_wrong_optimization_options : Error<
 def warn_drv_fsycl_add_default_spec_consts_image_flag_in_non_AOT : Warning<
   "-fsycl-add-default-spec-consts-image flag has an effect only in Ahead of Time Compilation mode (AOT)">,
   InGroup<SyclTarget>;
+def warn_drv_fsycl_device_obj_asm_device_only : Warning<
+  "-fsycl-device-obj=asm flag has an effect only when compiling device code "
+  "and emitting assembly, make sure both -fsycl-device-only and -S flags are "
+  "present; will be ignored">, InGroup<SyclTarget>;
```

Comment:
Only concern I have is the `InGroup` setting.  This associates the diagnostic with `-Wsycl-target` which is more geared towards problems with SYCL target specifications.  Use of `InGroup<UnusedCommandLineArgument>` may be more appropriate.